From 2ed58444c0ed3cbea77cb858d91ade6e5435f153 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 9 Nov 2018 17:20:27 +0800
Subject: [PATCH 5088/5242] MLK-20307 gpu: imx: dpu: extdst: Fix get_xval()

commit  92899d70e5c4a0a6a2c2fd609bf52593a6c0147c from
https://source.codeaurora.org/external/imx/linux-imx.git

The function get_xval() returns the C_XVAL and L_XVAL fields of
the CURPIXELCNT and LASTPIXELCNT registers.  They are 16bit and
sit in the low 16bit of the registers.  This patch fixes the way
we mask the register and read the fields out.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-extdst.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-extdst.c b/drivers/gpu/imx/dpu/dpu-extdst.c
index a278723..6eb6a65 100644
--- a/drivers/gpu/imx/dpu/dpu-extdst.c
+++ b/drivers/gpu/imx/dpu/dpu-extdst.c
@@ -48,7 +48,7 @@
 #define CURPIXELCNT			0x1C
 static u16 get_xval(u32 pixel_cnt)
 {
-	return pixel_cnt && 0xFF;
+	return pixel_cnt & 0xFFFF;
 }
 
 static u16 get_yval(u32 pixel_cnt)
-- 
1.7.9.5

